appeals/web/src/server/appeals/appeal-details/appellant-case/__tests__/appellant-case.test.js

```diff
@@ -1162,6 +1163,212 @@ describe('appellant-case', () => {
+				.send({
+					'due-date-day': '1',
+					'due-date-month': '13',
+					'due-date-year': '3000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+
+			response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '1',
+					'due-date-month': 'dec',
+					'due-date-year': '3000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the valid date page with the expected error message if an invalid year was provided', async () => {
+			let response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '1',
+					'due-date-month': '1',
+					'due-date-year': '23'
+				});
+
+			expect(response.statusCode).toBe(200);
+			let element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+
+			response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '1',
+					'due-date-month': '1',
+					'due-date-year': 'abc'
+				});
+
+			expect(response.statusCode).toBe(200);
+			element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the valid date page with the expected error message if an invalid date was provided', async () => {
+			const response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '29',
+					'due-date-month': '2',
+					'due-date-year': '3000'
+				});
+
+			expect(response.statusCode).toBe(200);
+			const element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the valid date page with the expected error message if date was in past but prior to the date the case was received', async () => {
+			const response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '20',
+					'due-date-month': '5',
+					'due-date-year': '2023'
+				});
+
+			expect(response.statusCode).toBe(200);
+			const element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should redirect to the check and confirm page if a valid date was provided', async () => {
+			const response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '22',
+					'due-date-month': '5',
+					'due-date-year': '2023'
+				});
+
+			expect(response.statusCode).toBe(200);
```

Comment:
Can you check it is redirecting to the right place? 
expect(response.text).toEqual(
				'Found. Redirecting to {urlhere}'
			);